### PR TITLE
Noisize Node

### PR DESCRIPTION
**Pull request by vicdoval**
image

Seed and scale per vertex can be passed, in this example the seed is chosen by determining the closest point of another mesh and the scale is based on the distance to that point
image

One matrix per point can be passed if the matrix list is wrapped, note that the "Flat Output" checkbox of the matrix in is un-checked
image

The node offers three ways of matching the list lengths "Repeat Last", "Cycle" and "Match Short" in this example "Cycle" is used to alternate the noise matrix
image

In this example the scale output is used to blend with another oscillation texture
image

The "Vector" Mode does not use vertex normals so it can be used just with vertices
image

The "Scale out" input can be used to mask the affected vertices
image

- [x] Code changes complete.
- [x] Code documentation complete.
- [x] Documentation for users complete (or not required, if user never sees these changes).
- [x] Manual testing done. 
- [] Unit-tests implemented.
- [x] Ready for merge.

### CODE REVIEW

**Comment by portnov** [comment]
Is this something like "Vector noise" + Move?

**Comment by vicdoval** [comment]
@portnov  talking short: yes
But it also pre-multiplies the vector by the inverted matrix (To use a blender empty to control the noise), allows seed per vertex and multiplies the output by a vector and gives the chance to apply it to the normal. Also with some candy list.machting :)

**Comment by zeffii** [comment]
"Apply Noise"  , good - i had a branch with something similar which can now be ditched :), but i did have a "rescale" option, to scale in the input , then get noise, then add noise to original input

thanks @vicdoval  :)

**Comment by vicdoval** [comment]
@zeffii i was following you attempt at that point :)
I started also with the pre-scale concept, then I added offset, then added rotatation... And finally i found it was more intuitive to invert the matrix and call it Noise Matrix as the Object UVMapping in the Displace Modifier.:)  (Btw i have started a Displace node to work with blender textures)

File: nodes/transforms/noisize.py [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+
+class SvNoisizeNode(bpy.types.Node, SverchCustomTreeNode):
+    """
+    Triggers: Noise Displacement
+    Tooltip: Move input verts/mesh with a noise values.
+
+    """
+
+    bl_idname = 'SvNoisizeNode'
+    bl_label = 'Noisize'
```
Comment (zeffii) [inline nodes/transforms/noisize.py]: this name / label  ,  could be better... "Noise Displacer" even.  Not a big issue, maybe Noisize feels a little gimmicky ?

Comment (vicdoval) [reply nodes/transforms/noisize.py]: It was recalling Randomize but it is true Noisize  is a little weird...

Comment (zeffii) [reply nodes/transforms/noisize.py]: "Perturb", "Add Noise"

Comment (vicdoval) [reply nodes/transforms/noisize.py]: Noise Displacer sounds fine to me, but would you change also the class name and the files names?

Comment (vicdoval) [reply nodes/transforms/noisize.py]: Maybe just "Noise Displace" is even cleaner

Comment (zeffii) [reply nodes/transforms/noisize.py]: yeah, but do it early.

**Comment by catzova** [comment]
where can I find plane mk2?

**Comment by Durman** [comment]
It's called Plane now.

**Pull request merged**